common/src/main/java/com/wynntils/models/horse/HorseModel.java

```diff
@@ -30,6 +31,19 @@ public Optional<HorseItem> getHorse() {
         return Models.Item.asWynnItem(McUtils.inventory().getItem(horseSlot), HorseItem.class);
     }
 
+    public Optional<CappedValue> calculateNextLevelMinutes() {
+        HorseItem horseItem = getHorse().get();
+
+        // This is based off of a formula from https://wynncraft.wiki.gg/wiki/Horses#Levels
+        double levelProgress = 3.0 * horseItem.getLevel().current() + 2;
+        double xpProgress = 100.0 - horseItem.getXp().current();
+
+        double result = levelProgress / 6.0 * (xpProgress / 100.0) * 100.0;
+        double resultMax = levelProgress / 6.0 * 100.0;
+
+        return Optional.of(new CappedValue((int) Math.ceil(result), (int) Math.ceil(resultMax)));
+    }
```

Comment:
You should return an empty optional, if there is no horse item, or if some data is missing and you can't calculate it. Sorry if I was not clear before.

Reply:
Sorry, made it return an empty optional when there's missing data.